test/framework/integration_framework/fake_peer/block_storage.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "framework/integration_framework/fake_peer/block_storage.hpp"
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/range/adaptor/map.hpp>
+#include <boost/range/algorithm/max_element.hpp>
+#include "backend/protobuf/block.hpp"
+#include "framework/integration_framework/fake_peer/fake_peer.hpp"
+
+/// Emplace to a map.
+/// @return true if overwritten.
+template <typename MapType, typename KeyType, typename ValueType>
+static bool emplaceCheckingOverwrite(MapType &map,
```

Comment:
wondering for moving to c++17 :( https://en.cppreference.com/w/cpp/container/map/insert_or_assign